src/attributes/classes.js

```diff
@@ -114,22 +114,31 @@ jQuery.fn.extend({
 		}
 
 		return this.each(function() {
-			var className, i, self, classNames;
+			var className, i, self, classNames, predicate;
 
 			if ( type === "string" ) {
 
 				// Toggle individual class names
 				i = 0;
 				self = jQuery( this );
+				predicate = jQuery.isFunction( stateVal ) ? stateVal.call( self, i, self) : null;
```

Comment:
Can we write the docs for this before landing it? I can see several different interpretations. Just for review, [we already have these](http://api.jquery.com/toggleClass/):
```js
toggleClass(String classNames)
toggleClass(String classNames, Boolean state)
toggleClass(Boolean state)
toggleClass(Function(index, currentClassNames, state), Boolean state)
```
In all cases the `state` is optional. so the proposal is to add `.toggleClass(String classNames, Function)` to the list. I can see that working several ways:

1. Function called once for the whole collection, given the incoming `jQuery` object collection as `this` and the String list of classes as an arg, returning a Boolean. In essence the function is just returning the `state` that `toggleClass(String classNames, Boolean state)` uses, so this option has that going for it. I think we want a return of `undefined` to specify the default toggling behavior?
2. Function called once for each element, given the element as `this` and the index as an arg. This is close to what the code above does, assuming we use the index from the `.each()` above it instead of the `i`. However, that means the function doesn't see the classes being toggled, wouldn't we also want to pass them in? The current PR already allows `undefined` to specify the default toggling behavior.
3. Function called for each combination of element and className, given the element as `this` and the index and className as args. More function calls but avoids the need for an explicit loop inside the called function if iterating over the classes is a common need. Again with the `undefined` for toggling.

Also note that undocumented variants like `toggleClass(Function, Function)` are going to start doing something now due to the way we look at `state` in the code, I don't know if those should be cause for concern or documentation.